src/DGtal/io/doc/moduleQGLExtension.dox

```diff
@@ -0,0 +1,85 @@
+
+
+/*
+ * Useful to avoid writing DGtal:: in front of every class.
+ */
+namespace DGtal {
+
+/**
+ * @page moduleQGLExtension Extending the QGLViewer Viewer3D interface
+ *  
+ *
+ * This part of the manual describes how to extend the standard
+ * Viewer3D interface of DGtal. For instance, you wish to assign new
+ * reactions to keys or to mouse event, or you wish to display or
+ * select other graphical elements. To do that, you just have to write
+ * a class that derived from Viewer3D::Extension and overrides some
+ * methods to get your desired behaviour.
+ *  
+ * @author Jacques-Olivier Lachaud
+ *
+ * @since 9.4
+ * 
+ * Related examples: viewer3D-11-extension.cpp
+ *
+ * [TOC]
+ *
+ * \section QGLExtension_sec1 Extending the standard behavior of Viewer3D
+ *
+ * The Viewer3D interface provides a very handy way of visualising 3D
+ * scenes, digital objects or meshes in DGtal, and is bundled with
+ * many functionalities (like moving the light, selecting some
+ * objects, etc). However it is not directly extensible as is, and,
+ * for several reasons, you cannot derive from the Viewer3D class
+ * directly to add some new functionalities.
+ *
+ * Therefore we provide the class Viewer3D::Extension so that you can
+ * extend the viewer in several ways. The principle is very easy. This
+ * interface provides a set of empty virtual methods so you just have
+ * to override in order to change the behavior of the viewer. For
+ * instance, you just have to override
+ * Viewer3D::Extension::keyPressEvent to capture other key events. You
+ * can override all the following methods:
+ *
+ * - Viewer3D::Extension::keyPressEvent. This method may be
+ *   overloaded to capture other key events.
+ *
+ * - Viewer3D::Extension::drawWithNames. This method is useful for
```

Comment:
It is used in the mini-interaction method, see for instance
https://github.com/DGtal-team/DGtal/blob/master/examples/io/viewers/viewer3D-10-interaction.cpp
I do not have a useful example here for overriding it, but it was natural to provide ways to override all the standard methods provided by libQGLViewer.